tests/functional/splinter_test.c

```diff
@@ -731,42 +761,23 @@ test_trunk_destroy_tables(trunk_handle   **spl_tables,
    platform_free(hid, spl_tables);
 }
 
-platform_status
-test_splinter_perf(trunk_config    *cfg,
-                   test_config     *test_cfg,
-                   allocator       *al,
-                   cache           *cc[],
-                   uint64           num_insert_threads,
-                   uint64           num_lookup_threads,
-                   uint64           num_range_threads,
-                   uint32           max_async_inflight,
-                   task_system     *ts,
-                   platform_heap_id hid,
-                   uint8            num_tables,
-                   uint8            num_caches,
-                   uint64           insert_rate)
+/*
+ * compute_per_table_inserts() --
+ *
+ * Helper function to compute the # of inserts for each table, based on the
+ * configuration specified.
+ *
+ * Returns: Total # of inserts to-be-done in the workload
+ */
+static uint64
+compute_per_table_inserts(uint64       *per_table_inserts, // OUT
+                          trunk_config *cfg,               // IN
+                          test_config  *test_cfg,          // IN
+                          uint8         num_tables)
```

Comment:
Here on, you will see chunks of code modularized out into sub-functions. No logic has been done, I assure you.